### PR TITLE
feat(payment): BOLT-23 added custom embedded credit card fields as Bolt payment method

### PR DESCRIPTION
**Pull request by Sireks**
## What?
Add Bolt embedded fields as Bolt custom credit card form in Bolt payment option (Checkout Payment step)

## Why?
Because of the task: https://jira.bigcommerce.com/browse/BOLT-23

## Sibling pull-request
checkout-sdk-js: https://github.com/bigcommerce/checkout-sdk-js/pull/1195

## Testing / Proof
Here is test result:
checkout-js-test

Here is how it looks like:
bolt-embedded-fields

### CODE REVIEW

**Review by BC-ACherednichenko — APPROVED** [review]

File: src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    const initializeBoltPayment = useCallback(options => initializePayment({
+            ...options,
+            bolt: {
+                containerId: boltEmbeddedContainerId,
+                useBigCommerceCheckout: true,
+            },
+        }
+    ), [initializePayment, boltEmbeddedContainerId]);
+
+    function renderCustomPaymentForm() {
```
Comment (davidchin) [inline src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx:23]: Can we define it as `useCallback` so that the prop is memoized?

File: src/app/payment/paymentMethod/getPaymentMethodName.ts [hunk 2]
```diff
@@ -56,6 +56,10 @@ export function getTranslatedPaymentMethodName(
             return language.translate('payment.amazon_name_text');
         }
 
+        if (method.id === PaymentMethodId.Bolt) {
+            return language.translate('payment.bolt_name_text');
```
Comment (davidchin) [inline src/app/payment/paymentMethod/getPaymentMethodName.ts:60]: https://app.circleci.com/pipelines/github/bigcommerce/checkout-js/3357/workflows/1a47ca2e-d97f-4aac-bf20-87d6256250dc/jobs/8428 :( This method started off with just a few `if` but now it has more than 10. Maybe it's a sign that we need to revisit the implementation. Perhaps maybe have a dictionary like

```tsx
const translations = {
    [PaymentMethodId.Bolt]: language.translate('payment.bolt_name_text'),
    [PaymentMethodId.Amazon]: language.translate('payment.amazon_name_text'),
    // ...
}
```
And then you can read the translation by `translations[method.id]`. That way, we can remove all the `if` and reduce the cyclomatic complexity.

File: src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx (hunk 1 above)
Comment (Sireks) [reply src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx:23]: Ok I will define it as `useCallback`

File: src/app/payment/paymentMethod/getPaymentMethodName.ts (hunk 2 above)
Comment (Sireks) [reply src/app/payment/paymentMethod/getPaymentMethodName.ts:60]: I think that it is a good idea

Comment (Sireks) [reply src/app/payment/paymentMethod/getPaymentMethodName.ts]: Changed

File: src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx (hunk 1 above)
Comment (Sireks) [reply src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx]: Changed

**Review by davidchin — APPROVED** [review]
Nice one, thanks @Sireks 👍

File: src/app/locale/translations/en.json [hunk 3]
```diff
@@ -204,6 +205,7 @@
             "checkoutcom_sepa_debtor_title": "Debtor",
             "checkoutcom_sepa_mandate_disclaimer": "By accepting this mandate form, you authorize {creditorName} to send instructions to your bank to debit your account, and your bank to debit your account in accordance with those instructions. You are entitled to a refund from your bank under the terms and conditions of your agreement with your bank. A refund must be claimed within 8 weeks starting from the date on which your account was debited.",
             "checkoutcom_sepa_payment_type": "Payment type: one-off (non-recurring)",
+            "clearpay_name_text": "Clear Pay",
```
Comment (TomA-R) [inline src/app/locale/translations/en.json:208]: Is this related?
